lib/rex/proto/kerberos/credential_cache/krb5_ccache_presenter.rb

```diff
@@ -0,0 +1,276 @@
+      end.join("\n")
+      output.join("\n")
+    end
+
+    protected
+
+    # @return [Rex::Proto::Kerberos::CredentialCache::Krb5Ccache]
+    attr_reader :ccache
+
+    # @param [Rex::Proto::Kerberos::CredentialCache::Krb5CcacheCredential] cred
+    # @param [String,nil] key Decryption key for the encrypted part
+    # @return [String] A human readable representation of a ccache credential
+    def present_cred(cred, key: nil)
+      output = []
+      output << "Server: #{cred.server}"
+      output << "Client: #{cred.client}"
+      output << "Ticket etype: #{cred.keyblock.enctype} (#{Rex::Proto::Kerberos::Crypto::Encryption.const_name(cred.keyblock.enctype)})"
+      output << "Key: #{cred.keyblock.data.unpack1('H*')}"
+      output << "Ticket Length: #{cred.ticket.length}"
+      ticket = Rex::Proto::Kerberos::Model::Ticket.decode(cred.ticket.value)
+
+      output << "Subkey: #{cred.is_skey == 1}"
+      output << "Addresses: #{cred.address_count}"
+
+      unless cred.address_count == 0
+        output << cred.addresses.map do |address|
+          "#{ADDRESS_TYPE_MAP.fetch(address.addrtype, address.addrtype)}: #{address.data}".indent(2)
+        end.join("\n")
+      end
+
+      output << "Authdatas: #{cred.authdata_count}"
+      unless cred.authdata_count == 0
+        output << cred.authdatas.map do |authdata|
+          "#{AD_TYPE_MAP.fetch(authdata.adtype, authdata.adtype)}: #{authdata.data}".indent(2)
+        end.join("\n")
+      end
+
+      output << 'Times:'
+      output << "Auth time: #{cred.authtime}".indent(2)
+      output << "Start time: #{cred.starttime}".indent(2)
+      output << "End time: #{cred.endtime}".indent(2)
+      output << "Renew Till: #{cred.renew_till}".indent(2)
+
+      output << 'Ticket:'
+      output << "Ticket Version Number: #{ticket.tkt_vno}".indent(2)
+      output << "Realm: #{ticket.realm}".indent(2)
+      output << "Server Name: #{ticket.sname}".indent(2)
+      output << 'Encrypted Ticket Part:'.indent(2)
+      output << "Ticket etype: #{ticket.enc_part.etype} (#{Rex::Proto::Kerberos::Crypto::Encryption.const_name(ticket.enc_part.etype)})".indent(4)
+      output << "Key Version Number: #{ticket.enc_part.kvno}".indent(4)
+
+      if key.blank?
+        output << 'Cipher:'.indent(4)
+        output << Base64.strict_encode64(ticket.enc_part.cipher).indent(6)
+      else
+        output << "Decrypted (with key: #{key.bytes.map { |x| "\\x#{x.to_s(16).rjust(2, '0')}" }.join}):".indent(4)
+        output << present_encrypted_ticket_part(ticket, key).indent(6)
+      end
+
+      output.join("\n")
+    end
+
+    # @param [Rex::Proto::Kerberos::Pac::Krb5PacInfoBuffer] info_buffer
+    # @return [String] A human readable representation of a Logon Information
+    def present_logon_info(info_buffer)
+      validation_info = info_buffer.buffer.pac_element.data
+
+      output = []
+      output << 'Validation Info:'
```

Comment:
Not a blocker; I'm not sure if it would be more readable or not - but this could also have used heredoc syntax

```ruby
output = <<~EOF
  Validation Info:
    Logon Time: #{present_time(validation_info.logon_time)}
    Logoff Time: #{present_time(validation_info.logoff_time)}
    ... etc ...
    Group Ids:
      #{present_group_ids(validation_info).indent(2)}
    ... etc ...
EOF
```